include/ensmallen_bits/cmaes/weight_init_policies/default_weight.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file default_weight.hpp
+ * @author John Hoang 
+ * 
+ * Default initialization weight
+ * 
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ * 
+ */
+#ifndef ENSMALLEN_CMAES_DEFAULT_WEIGHT_HPP
+#define ENSMALLEN_CMAES_DEFAULT_WEIGHT_HPP
+
+namespace ens{
+
+class DefaultWeight{
+
+ public:
+  /**
+   * Constructor
+   * 
+   */
+  DefaultWeight()
+  {
+    // Doing nothing
+  }
+
+  /**
+   * This function will generate raw weights and mu_eff first 
+   * 
+   * @param lambda The length of raw weights 
+   */
+  void GenerateRaw(const size_t lambda)
+  {
+    len = lambda;
+    mu = std::round(len/2);
+    weights = std::log(mu + 0.5) - arma::log(arma::linspace<arma::Row<double> >(0, len - 1, len) + 1.0);
+    for(size_t i = mu; i < len; ++i)
+    {
+      weights(i) = 0;
+    }
+    assert(weights(mu-1) > 0);
+    double sumPos = arma::accu(weights.cols(0, mu-1));
+    // positive weights sum to one
+    for(size_t i = 0; i < mu; ++i)
+    {
+      weights(i) /= sumPos;
+    }
+    mu_eff = 1 / arma::accu(arma::pow(weights.cols(0, mu-1), 2)); 
```

Comment:
Can you use camel casing for all names, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more details.